pkg/bpmn_engine/engine_timer.go

```diff
@@ -36,12 +40,11 @@ func (state *BpmnEngineState) handleIntermediateTimerCatchEvent(process *Process
 	return false
 }
 
-func createNewTimer(process *ProcessInfo, instance *ProcessInstanceInfo, ice BPMN20.TIntermediateCatchEvent, generateKey func() int64) *Timer {
+func createNewTimer(process *ProcessInfo, instance *ProcessInstanceInfo, ice BPMN20.TIntermediateCatchEvent,
+	generateKey func() int64) (*Timer, error) {
 	durationVal, err := findDurationValue(ice, process)
 	if err != nil {
-		msg := fmt.Sprintf("Error parsing 'timeDuration' value from element with ID=%s. Error:%s", ice.Id, err.Error())
-		// TODO: should not panic
-		panic(msg)
+		return nil, BpmnValidationError
```

Comment:
Good idea with using the variable. Since we will use it more often and also provide more details to the users, we should provide additional validation error information like this

```suggestion
		return nil, &BpmnValidationError{Msg: fmt.Sprintf("Error parsing 'timeDuration' value "+
			"from element with ID=%s. Error:%s", ice.Id, err.Error())}
```